addons/l10n_ro_edi_etransport/models/l10n_ro_edi_etransport_document.py

```diff
@@ -0,0 +1,563 @@
+]
+
+SCHEMATRON_ERROR_ID_PATTERN = r'BR-(?:CL-)?\d{3}'
+
+ETRANSPORT_TEST_URL = 'https://api.anaf.ro/test/ETRANSPORT/ws/v1'
+ETRANSPORT_PROD_URL = 'https://api.anaf.ro/prod/ETRANSPORT/ws/v1'
+
+OPERATION_TYPE_TO_ALLOWED_SCOPE_CODES = {
+    "10": ("101", "201", "301", "401", "501", "601", "703", "801", "802", "901", "1001", "1101", "9901"),
+    "20": ("101", "301", "703", "801", "802", "9901"),
+    "30": ("101", "704", "705", "9901"),
+}
+
+LOCATION_TYPE_MAP = {
+    'start': {
+        'customs_code': '40',
+        'bcp_codes': ('10', '12', '14', '60'),
+    },
+    'end': {
+        'customs_code': '50',
+        'bcp_codes': ('10', '20', '22', '24', '70')
+    }
+}
+
+BORDER_CROSSING_POINTS = [
+    ('1', "Petea (HU)"),
+    ('2', "Borș(HU)"),
+    ('3', "Vărșand(HU)"),
+    ('4', "Nădlac(HU)"),
+    ('5', "Calafat (BG)"),
+    ('6', "Bechet(BG)"),
+    ('7', "Turnu Măgurele(BG)"),
+    ('8', "Zimnicea(BG)"),
+    ('9', "Giurgiu(BG)"),
+    ('10', "Ostrov(BG)"),
+    ('11', "Negru Vodă(BG)"),
+    ('12', "Vama Veche(BG)"),
+    ('13', "Călărași(BG)"),
+    ('14', "Corabia(BG)"),
+    ('15', "Oltenița(BG)"),
+    ('16', "Carei  (HU)"),
+    ('17', "Cenad (HU)"),
+    ('18', "Episcopia Bihor (HU)"),
+    ('19', "Salonta (HU)"),
+    ('20', "Săcuieni (HU)"),
+    ('21', "Turnu (HU)"),
+    ('22', "Urziceni (HU)"),
+    ('23', "Valea lui Mihai (HU)"),
+    ('24', "Vladimirescu (HU)"),
+    ('25', "Porțile de Fier 1 (RS)"),
+    ('26', "Naidăș(RS)"),
+    ('27', "Stamora Moravița(RS)"),
+    ('28', "Jimbolia(RS)"),
+    ('29', "Halmeu (UA)"),
+    ('30', "Stânca Costești (MD)"),
+    ('31', "Sculeni(MD)"),
+    ('32', "Albița(MD)"),
+    ('33', "Oancea(MD)"),
+    ('34', "Galați Giurgiulești(MD)"),
+    ('35', "Constanța Sud Agigea"),
+    ('36', "Siret  (UA)"),
+    ('37', "Nădlac 2 - A1 (HU)"),
+    ('38', "Borș 2 - A3 (HU)"),
+]
+
+STATE_CODES = {
+    'AB': '1',
+    'AR': '2',
+    'AG': '3',
+    'BC': '4',
+    'BH': '5',
+    'BN': '6',
+    'BT': '7',
+    'BV': '8',
+    'BR': '9',
+    'BZ': '10',
+    'CS': '11',
+    'CJ': '12',
+    'CT': '13',
+    'CV': '14',
+    'DB': '15',
+    'DJ': '16',
+    'GL': '17',
+    'GJ': '18',
+    'HR': '19',
+    'HD': '20',
+    'IL': '21',
+    'IS': '22',
+    'IF': '23',
+    'MM': '24',
+    'MH': '25',
+    'MS': '26',
+    'NT': '27',
+    'OT': '28',
+    'PH': '29',
+    'SM': '30',
+    'SJ': '31',
+    'SB': '32',
+    'SV': '33',
+    'TR': '34',
+    'TM': '35',
+    'TL': '36',
+    'VS': '37',
+    'VL': '38',
+    'VN': '39',
+    'B': '40',
+    'CL': '51',
+    'GR': '52',
+}
+
+SELECTION_LOC_TYPE_1 = [('location', "Location"), ('bcp', "Border Crossing Point")]
+SELECTION_LOC_TYPE_2 = [('location', "Location"), ('bcp', "Border Crossing Point"), ('customs', "Customs Office")]
+
+
+class L10nRoEdiETransportDocument(models.Model):
+    _name = 'l10n_ro.edi.etransport.document'
+    _description = "Romanian eTransport document"
+    _order = 'create_date DESC, id DESC'
```

Comment:
This would not make sense: the eFactura document requires a relationship with an invoice which does not make sense in the context of the eTransport flow.

Reply:
For now, I guess we don't have to because they're still completely separate, but in the future we'll need to do that. The `l10n_ro_edi.document` object is created/designed with this in mind. Maybe in master later we can change the `required` attribute for `invoice_id`, and use `auto_join`/etc (idk which attribute it is) so that only *either* invoice/picking/(or other) is required.